packages/neo-one-node-core/src/ConsensusData.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  SerializableWire,
+  SerializableJSON,
+  DeserializeWireBaseOptions,
+  SerializeWire,
+  createSerializeWire,
+} from './Serializable';
+import { ConsensusDataJSON, BinaryWriter, IOHelper, crypto, UInt256 } from '@neo-one/client-common';
+import { BN } from 'bn.js';
+import { utils } from './utils';
+
+// TODO: could put this somewhere else it is possible it changes?
+const defaultValidatorsCount = new BN(7);
+
+export interface ConsensusDataAdd {
+  readonly primaryIndex: number;
+  readonly nonce: BN;
+}
+
+export class ConsensusData implements SerializableWire<ConsensusData>, SerializableJSON<ConsensusDataJSON> {
+  public static deserializeWireBase(options: DeserializeWireBaseOptions): ConsensusData {
+    const { reader } = options;
+    const primaryIndex = reader.readVarUIntLE(defaultValidatorsCount.subn(1)).toNumber();
+    const nonce = reader.readUInt64LE();
+
+    return new this({
+      primaryIndex,
+      nonce,
+    });
+  }
+
+  public readonly primaryIndex: number;
+  public readonly nonce: BN;
+  public readonly serializeWire: SerializeWire = createSerializeWire(this.serializeWireBase.bind(this));
+  public readonly sizeInternal = utils.lazy(
+    () => IOHelper.sizeOfVarUIntLE(this.primaryIndex) + IOHelper.sizeOfUInt64LE,
+  );
+  public readonly hashInternal = utils.lazy(() => crypto.hash256(this.serializeWire()));
+
+  public constructor({ primaryIndex, nonce }: ConsensusDataAdd) {
+    this.primaryIndex = primaryIndex;
+    this.nonce = nonce;
```

Comment:
`this.hashInternal = hashIn === undefined ? utils.lazy(() => crypto.hash256(this.message)) : () => hashIn`   ?

See `ConsensusData.ts` here: https://github.com/neo-one-suite/neo-one/pull/2131/files#diff-eaa1ebfacb78fd619fd18a95571c3aa4